geth/rpc/client.go

```diff
@@ -72,8 +84,91 @@ func (c *Client) Call(result interface{}, method string, args ...interface{}) er
 //
 // It uses custom routing scheme for calls.
 func (c *Client) CallContext(ctx context.Context, result interface{}, method string, args ...interface{}) error {
+	// check locally registered handlers first
+	if handler, ok := c.handler(method); ok {
+		return c.callMethod(ctx, result, handler, args)
+	}
+
 	if c.router.routeRemote(method) {
 		return c.upstream.CallContext(ctx, result, method, args...)
 	}
 	return c.local.CallContext(ctx, result, method, args...)
 }
+
+// RegisterHandler registers local handler for specific RPC method.
+//
+// If method is registered, it will be executed with given handler and
+// never routed to the upstream or local servers.
+func (c *Client) RegisterHandler(method string, handler Handler) {
+	c.mx.Lock()
+	defer c.mx.Unlock()
+
+	c.handlers[method] = handler
+}
+
+// callMethod calls registered RPC handler with given args and pointer to result.
+// It handles proper params and result converting
+//
+// TODO(divan): use cancellation via context here?
+func (c *Client) callMethod(ctx context.Context, result interface{}, handler Handler, args ...interface{}) error {
+	response, err := handler(ctx, args...)
+	if err != nil {
+		return err
+	}
+
+	// if result is nil, just ignore result
```

Comment:
@tiabc hmm, I remember that saw in docs, that response without result is totally valid JSON-RPC, but now quick check revealed that `result` is required. I'll double check it, because I remember clearly reading this in some doc (hence the comment, emphasizing that it's ok).

Reply:
In theory, it's possible to call the RPC client like `client.CallContext(ctx, nil, "eth_sendRawTransaction", data)`, thus handling `nil` case.

Reply:
Found: I've read about it in rpc.CallContext comment:

> You can also pass nil, in which case the result is ignored.

 https://github.com/status-im/status-go/blob/develop/vendor/github.com/ethereum/go-ethereum/rpc/client.go#L248

Reply:
Clear! Let's add a comment?